core/src/main/scala/org/bitcoins/core/protocol/script/ScriptWitness.scala

```diff
@@ -106,6 +110,21 @@ sealed abstract class P2WSHWitnessV0 extends ScriptWitnessV0 {
     relevantStack.map(ECDigitalSignature.fromBytes)
   }
 
+  // Note that this is not guaranteed to work for non-standard script signatures
+  lazy val scriptSignature: ScriptSignature = {
+    val asm = stack.toVector.tail.reverse.flatMap { bytes =>
+      if (bytes.isEmpty) {
+        Vector(OP_0)
+      } else if (bytes.length == 1 && bytes.head <= 16 && bytes.head >= -1) {
```

Comment:
Maybe just

```scala
val numOpt = bytes.headOption.flatMap(byte => ScriptNumberOperation.fromNumber(byte))

...

else if (numOpt.isDefined)) { 
  Vector(numOpt.get)
} 
...

Reply:
I personally like how it is because it clearly enumerates all cases